nuxdie/blog
# Update dependency react-dates to version 9.x

This Pull Request updates dependency react-dates from version `8.1.1` to `9.0.1`

### Changelog

#### 9.0.1 / 2017-04-04

  * Version 9.0.1
  * Fix DateInput tests for karma
  * [Tests] &#x60;DayPicker&#x60;: cover more code by enabling lifecycle methods
  * [Tests] &#x60;DateInput&#x60;: cover &#x60;componentDidUpdate&#x60;
  * [Tests] &#x60;CalendarMonthGrid&#x60;: &#x60;componentWillReceiveProps&#x60;: cover when numberOfMonths does not change
  * [Tests] consolidate &#x60;DayPicker.defaultProps.initialVisibleMonth&#x60; definition.
  * [Tests] drop “lines” coverage; 82 isn’t 81.99 :-p
  * [greenkeeper] ignore &#x60;webpack&#x60;
    Closes [#401](https://github.com/airbnb/react-dates/issues/401)
  * [Tests] increase test coverage
  * [Tests] ensure karma tests don’t also run regular tests.
  * [Tests] ensure &#x60;npm run cover&#x60; has enough RAM to complete.
  * [Fix] &#x60;OutsideClickHandler&#x60;: remove useless &#x60;forbidExtraProps&#x60; call on &#x60;defaultProps&#x60;
    - add TODO to forbidExtraProps on &#x60;propTypes&#x60;, semver-major
  * Merge pull request [#421](https://github.com/airbnb/react-dates/issues/421) from airbnb/maja-fix-firefox-outside-click-with-portal-woes
    Uses React&#x27;s onClick handler instead of &#x60;OutsideClickHandler&#x60; with &#x60;withPortal&#x60; implementations
  * Merge pull request [#419](https://github.com/airbnb/react-dates/issues/419) from dburles/patch-1
    update DayPicker story to show single month
  * Uses React&#x27;s onClick handler instead of &#x60;OutsideClickHandler&#x60; with &#x60;withPortal&#x60; implementations
    This fixes an issue with Firefox specifically where the order of operations of the events caused the &#x60;withPortal&#x60; DayPicker to immediately close upon opening.
  * add 3 months story to DayPicker
    undo removing numberOfMonths prop from other stories
  * update DayPicker story to show single month
  * [Tests] fix coveralls
  * [Tests] improvements:
    - &#x60;node&#x60; &#x60;v7&#x60;/&#x60;v4.6&#x60;; drop &#x60;v5&#x60;
    - re-enable karma builds, but allow them to fail
    - correct before_script settings for karma tests
    - install appropriate react version, per &#x60;$REACT&#x60;
    - Ensure tests aren’t ran twice

#### 9.0.0 / 2017-03-31

  * Version 9.0.0
  * Fix modifer optimization specs by restoring sinon
  * Merge pull request [#406](https://github.com/airbnb/react-dates/issues/406) from moonboots/maja-add-calendarday-size-prop
    DaySize update
  * Adds daySize prop to the DRP/SDP/DP that scales everything accordingly
  * Merge pull request [#410](https://github.com/airbnb/react-dates/issues/410) from airbnb/maja-fix-done-bottom-bar-on-ios
    Separates out focusedInput from DOM Focus and focuses on the calendar for withPortal implementations
  * Separates out focusedInput from DOM Focus and focuses on the calendar for withPortal implementations
  * Merge pull request [#415](https://github.com/airbnb/react-dates/issues/415) from moonboots/optimize-is-same-day
    Optimize isSameDay
  * Optimize isSameDay
  * Merge pull request [#414](https://github.com/airbnb/react-dates/issues/414) from moonboots/jack-tap-perf-fixes-specs
    Tap performance followup, fixes and specs
  * Specs
  * Revert &quot;Speed up isSameDay&quot;
    This reverts commit d2e2320974aa2047b800d0be1e42d1780341e644.
  * Merge pull request [#412](https://github.com/airbnb/react-dates/issues/412) from moonboots/jack-tap-perf
    Improve touch tap performance
  * Speed up isSameDay
  * Shortcircuit for start and end date dependent modifiers
  * Avoid computing hover modifiers for touch
  * Changes storybook title text and link using setOptions addon
  * Merge pull request [#407](https://github.com/airbnb/react-dates/issues/407) from airbnb/greenkeeper/style-loader-0.16.1
    Update style-loader to the latest version 🚀
  * chore(package): update style-loader to version 0.16.1
    https://greenkeeper.io/
  * Merge pull request [#387](https://github.com/airbnb/react-dates/issues/387) from airbnb/maja-update-react-dates-documentation
    Simplify the README
  * Simplify the README
  * Merge pull request [#388](https://github.com/airbnb/react-dates/issues/388) from airbnb/greenkeeper/style-loader-0.15.0
    Update style-loader to the latest version 🚀
  * chore(package): update style-loader to version 0.15.0
    https://greenkeeper.io/
  * [Dev Deps] unpin &#x60;eslint-plugin-react&#x60;
    - fixed in https://github.com/yannickcr/eslint-plugin-react/issues/1117#issuecomment-287910318
    - [#383](https://github.com/airbnb/react-dates/issues/383)
  * [Dev Deps] pin &#x60;eslint-plugin-react&#x60; to v6.10.0.
    Closes [#383](https://github.com/airbnb/react-dates/issues/383). Related to https://github.com/yannickcr/eslint-plugin-react/issues/1117.
  * [Deps] pin &#x60;moment&#x60; to &lt; &#x60;v2.18&#x60;, since they don’t consider locale changes breaking, and our tests depend on the locale data.
    Closes [#381](https://github.com/airbnb/react-dates/issues/381); related to https://github.com/moment/moment/issues/3849

#### 8.2.1 / 2017-03-17

  * Version 8.2.1
  * Merge pull request [#379](https://github.com/airbnb/react-dates/issues/379) from airbnb/maja-update-dependencies-keep-greenkeeper-happy
    Update non-breaking dependencies
  * Update non-breaking dependencies
  * Merge pull request [#372](https://github.com/airbnb/react-dates/issues/372) from airbnb/greenkeeper/style-loader-0.14.0
    Update style-loader to the latest version 🚀
  * Merge pull request [#373](https://github.com/airbnb/react-dates/issues/373) from airbnb/greenkeeper/sinon-2.0.0
    Update sinon to the latest version 🚀
  * Merge pull request [#377](https://github.com/airbnb/react-dates/issues/377) from moonboots/disable-fastclick
    Disable fastclick
  * Disable fastclick
  * chore(package): update sinon to version 2.0.0
    https://greenkeeper.io/
  * chore(package): update style-loader to version 0.14.0
    https://greenkeeper.io/

#### 8.2.0 / 2017-03-14

  * Version 8.2.0
  * Merge pull request [#341](https://github.com/airbnb/react-dates/issues/341) from kupibilet-frontend/calendar-caption
    Calendar caption with renderCalendarCaption
  * Rename renderCalendarCaption to renderCalendarInfo
  * Introduced calendar caption with renderCalendarCaption
